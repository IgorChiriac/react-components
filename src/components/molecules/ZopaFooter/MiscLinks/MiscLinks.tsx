import React from 'react';
import styled from 'styled-components';

import FlexCol from '../../../layout/FlexCol/FlexCol';
import FlexRow from '../../../layout/FlexRow/FlexRow';
import FooterLink from '../FooterLink/FooterLink';

export interface IMiscLinkData {
  label: string;
  href: string;
}

const MiscLinkWrapper = styled.div`
  margin-bottom: 16px;
`;

const miscLinksData: IMiscLinkData[] = [
  {
    href: 'https://zopa.com',
    label: 'Homepage',
  },
  {
    href: 'https://zopa.com',
    label: 'Sitemap',
  },
  {
    href: 'https://zopa.com/privace-policy',
    label: 'Privacy policy',
  },
  {
    href: 'https://zopa.com/principles',
    label: 'Principles',
  },
  {
    href: 'https://zopa.com/cookie-policy',
    label: 'Cookie policy',
  },
  {
    href: 'https://zopa.com/conflicts-policy',
    label: 'Conflicts policy',
  },
  {
    href: 'https://zopa.com/website-terms',
    label: 'Website terms',
  },
];

const MiscLinks = () => (
  <FlexRow>
    {miscLinksData.map(({ label, href }, index) => (
      <FlexCol xs="auto" key={`footer-misc-links-${index}`}>
        <MiscLinkWrapper>
          <FooterLink size="small" href={href}>
            {label}
          </FooterLink>
        </MiscLinkWrapper>
      </FlexCol>
    ))}
  </FlexRow>
);

export default MiscLinks;
