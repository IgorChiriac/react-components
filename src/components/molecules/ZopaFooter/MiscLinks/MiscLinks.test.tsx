import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'react-testing-library';
import MiscLinks from './MiscLinks';

describe('<MiscLinks />', () => {
  it('renders the component', async () => {
    const { container } = render(<MiscLinks />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the component with no a11y violations', async () => {
    const { container } = render(<MiscLinks />);
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
