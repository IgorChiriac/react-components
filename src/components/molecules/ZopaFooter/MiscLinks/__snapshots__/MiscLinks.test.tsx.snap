// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MiscLinks /> renders the component 1`] = `
.c1 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: 16px;
  padding-left: 16px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: -16px;
  margin-left: -16px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c4 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  color: #4A3EDE;
}

.c4:hover {
  opacity: 0.88;
}

.c4:active {
  opacity: 0.72;
}

.c3 {
  color: #A9ACBA;
}

.c3:hover {
  color: #6F748B;
}

.c2 {
  margin-bottom: 16px;
}

@media (min-width:0px) {
  .c1 {
    display: block;
    -webkit-flex: 0 0 auto;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
    width: auto;
    max-width: none;
  }
}

<div
  class="c0"
  cols="12"
  direction="row"
>
  <div
    class="c1"
    cols="12"
  >
    <div
      class="c2"
    >
      <a
        class="c3 c4"
        href="https://zopa.com"
      >
        Homepage
      </a>
    </div>
  </div>
  <div
    class="c1"
    cols="12"
  >
    <div
      class="c2"
    >
      <a
        class="c3 c4"
        href="https://zopa.com"
      >
        Sitemap
      </a>
    </div>
  </div>
  <div
    class="c1"
    cols="12"
  >
    <div
      class="c2"
    >
      <a
        class="c3 c4"
        href="https://zopa.com/privace-policy"
      >
        Privacy policy
      </a>
    </div>
  </div>
  <div
    class="c1"
    cols="12"
  >
    <div
      class="c2"
    >
      <a
        class="c3 c4"
        href="https://zopa.com/principles"
      >
        Principles
      </a>
    </div>
  </div>
  <div
    class="c1"
    cols="12"
  >
    <div
      class="c2"
    >
      <a
        class="c3 c4"
        href="https://zopa.com/cookie-policy"
      >
        Cookie policy
      </a>
    </div>
  </div>
  <div
    class="c1"
    cols="12"
  >
    <div
      class="c2"
    >
      <a
        class="c3 c4"
        href="https://zopa.com/conflicts-policy"
      >
        Conflicts policy
      </a>
    </div>
  </div>
  <div
    class="c1"
    cols="12"
  >
    <div
      class="c2"
    >
      <a
        class="c3 c4"
        href="https://zopa.com/website-terms"
      >
        Website terms
      </a>
    </div>
  </div>
</div>
`;
